### DISCUSSION TITLE
Parser progress

### DISCUSSION THREAD

**Discussion by mark-wiemer** [zero-plusplus/vscode-autohotkey-debug#172] [Ideas]
Hiya, jumping in here to ask about the parser that you mentioned writing. I'm curious, have you considered looking at the source code for AHK itself and copying the built-in parser there? Seems like the logical starting place :)

I'm leading development for AutoHotkey Plus Plus, and we also need parser logic. Doesn't make sense to duplicate our efforts.

Do let me know what progress you've made.

**Comment by zero-plusplus** [comment]
**I use [DeepL translation](https://www.deepl.com/translator).**

I haven't learned C++, so I haven't looked at the AHK source code specifically.
Therefore, I wrote it based on experience and documentation.

Unfortunately, there are no plans to release the parser, it will just be included in an my extension to be released later.

This is because the parser is the most important part of my plan, and it is very difficult to define its specification.

This means that as the project progresses, there will be major specification changes if problems arise.

Therefore, when I may publish it, it will be when all my plans are complete and the specification is stable.

**Comment by zero-plusplus** [comment]
I answered your question with a guess, but I'm asking it to find out exactly what your question is.

> I'm leading development for AutoHotkey Plus Plus, and we also need parser logic. Doesn't make sense to duplicate our efforts.
> Do let me know what progress you've made.

What is the purpose of the question you are asking?

I guessed the following three things:
1. You want to borrow my parser
2. You want to integrate my plans into your extension
3. You are simply curious about the parser

I responded to the above answer assuming it was 1.
